src/graph.py

```diff
@@ -581,6 +582,45 @@ def grid_search(self, prize_file, Ws, Bs, Gs):
         return self.grid_randomization(prize_file, Ws, Bs, Gs, 0, 0)
 
 
+    ###########################################################################
+                #######          Summarize randomization results          #######
+    ###########################################################################
+
+    def generate_basic_statistics(self, robust_results): 
+        """
+        Summarizes robust network randomization results
+
+        Arguments: 
+            robust_results (dict of networkx): {paramstring: robust network (networkx)}
+
+        Returns: 
+            pd.DataFrame: summary of each robust network
+        """
+        robust_summary = {}
+
+        for paramstring, robust_network in robust_results.items(): 
+        
+            if robust_network.number_of_nodes() != 0: 
+                
+                robust_df = get_networkx_graph_as_dataframe_of_nodes(robust_network)
+                robust_summary[paramstring] = {
+                    "W": paramstring.split("_")[1],
+                    "B": paramstring.split("_")[3],
+                    "G": paramstring.split("_")[5],
+                    "size": len(robust_df), 
+                    "min_robustness":   robust_df.robustness.min(), 
+                    "mean_robustness":  robust_df.robustness.mean(), 
+                    "max_specificity":  robust_df.specificity.max(),
+                    "mean_specificity": robust_df.specificity.mean(), 
```

Comment:
please un-justify these (remove extra whitespace) or justify the entire dict. I'll leave it up to your preference.

Reply:
Done